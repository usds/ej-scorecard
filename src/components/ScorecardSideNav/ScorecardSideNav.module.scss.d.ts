declare namespace ScorecardSideNavNamespace {
  export interface IScorecardSideNavScss {
    scorecardSideNavContainer: string;
  }
}

declare const ScorecardSideNavScssModule: ScorecardSideNavNamespace.IScorecardSideNavScss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: ScorecardSideNavNamespace.IScorecardSideNavScss;
};

export = ScorecardSideNavScssModule;
