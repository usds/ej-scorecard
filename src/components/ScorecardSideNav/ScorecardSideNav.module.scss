// @use "uswds-core" as *;

// .scorecardSideNavContainer{

// }
