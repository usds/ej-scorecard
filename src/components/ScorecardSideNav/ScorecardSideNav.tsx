import React from 'react';
import { SideNav } from '@trussworks/react-uswds';

// import * as styles from './ScorecardSideNav.module.scss';

import { ScorecardSideNavProps } from '@/types';
import { groupAgencyLinks } from '../AppHeader/AppHeader';
import {
  AGENCY_NAME_RANGE1,
  AGENCY_NAME_RANGE2,
  AGENCY_NAME_RANGE3,
} from '@/data/constants';
import { isPathInGroupRange } from '../util';

const ScorecardSideNav: React.FC<ScorecardSideNavProps> = ({
  allAgencyNames,
  pathname,
}) => {
  if (isPathInGroupRange(pathname, AGENCY_NAME_RANGE1)) {
    return (
      <SideNav items={groupAgencyLinks(allAgencyNames, AGENCY_NAME_RANGE1)} />
    );
  } else if (isPathInGroupRange(pathname, AGENCY_NAME_RANGE2)) {
    return (
      <SideNav items={groupAgencyLinks(allAgencyNames, AGENCY_NAME_RANGE2)} />
    );
  } else if (isPathInGroupRange(pathname, AGENCY_NAME_RANGE2)) {
    return (
      <SideNav items={groupAgencyLinks(allAgencyNames, AGENCY_NAME_RANGE3)} />
    );
  } else {
    return null;
  }
};

export default ScorecardSideNav;
