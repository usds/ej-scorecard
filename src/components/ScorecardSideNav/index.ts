import ScorecardSideNav from './ScorecardSideNav';
export default ScorecardSideNav;
