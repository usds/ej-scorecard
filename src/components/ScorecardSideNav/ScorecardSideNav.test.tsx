import createComponentWithIntl from '@/test/testHelpers';

import ScorecardSideNav from './ScorecardSideNav';

export const mockAllAgencyNames = [
  `A1 Agency Dept Office`,
  `A2 Agency Dept Office`,
  `D1 Agency Dept Office`,
  `D2 Agency Dept Office`,
  `D3 Agency Dept Office`,
  `D4 Agency Dept Office`,
  `D5 Agency Dept Office`,
  `D6 Agency Dept Office`,
  `D7 Agency Dept Office`,
  `D8 Agency Dept Office`,
  `E1 Agency Dept Office`,
  `G1 Agency Dept Office`,
  `H1 Agency Dept Office`,
  `N1 Agency Dept Office`,
  `N2 Agency Dept Office`,
  `T1 Agency Dept Office`,
  `T2 Agency Dept Office`,
  `U1 Agency Dept Office`,
  `V1 Agency Dept Office`,
];
describe(`rendering of ScorecardSideNav Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <ScorecardSideNav
        allAgencyNames={mockAllAgencyNames}
        pathname={`scorecard/d6-agency-dept-office`}
      />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
