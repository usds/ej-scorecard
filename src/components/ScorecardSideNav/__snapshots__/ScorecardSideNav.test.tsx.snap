// Vitest Snapshot v1

exports[`rendering of ScorecardSideNav Component > checks if component renders 1`] = `null`;
